modular_skyrat/modules/bluespace_miner/code/bluespace_miner.dm

```diff
@@ -0,0 +1,167 @@
+	for(var/obj/machinery/bluespace_miner/bs_miner in range(1, src))
+		if(bs_miner != src)
+			. += span_warning("[src] is in a suboptimal environment: TOO CLOSE TO ANOTHER BLUESPACE MINER")
+
+/obj/machinery/bluespace_miner/proc/check_factors()
+	if(!COOLDOWN_FINISHED(src, process_speed))
+		return FALSE
+	COOLDOWN_START(src, process_speed, processing_speed)
+	// cant be broken or unpowered
+	if(machine_stat & (NOPOWER|BROKEN))
+		return FALSE
+	// cant be unanchored or open panel
+	if(!anchored || panel_open)
+		return FALSE
+	for(var/obj/machinery/bluespace_miner/bs_miner in range(1, src))
+		if(bs_miner != src)
+			return FALSE
+	var/turf/src_turf = get_turf(src)
+	var/datum/gas_mixture/environment = src_turf.return_air()
+	// if its hotter than (or equal to) room temp, don't work
+	if(environment.temperature >= T20C)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		return FALSE
+	// if its lesser than(or equal to) normal pressure, don't work
+	if(environment.return_pressure() <= ONE_ATMOSPHERE)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		return FALSE
+	// overpressurizing will cause nuclear particles...
+	if(environment.return_pressure() >= (ONE_ATMOSPHERE * 1.5))
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		return FALSE
+	//add amount_produced degrees to the temperature
+	var/datum/gas_mixture/merger = new
+	merger.assert_gas(/datum/gas/carbon_dioxide)
+	merger.gases[/datum/gas/carbon_dioxide][MOLES] = MOLES_CELLSTANDARD
+	merger.temperature = (T20C + gas_temp)
+	src_turf.assume_air(merger)
+	return TRUE
+
+/obj/machinery/bluespace_miner/proc/spawn_mats()
+	var/obj/chosen_sheet = pick_weight(ore_chance)
+	new chosen_sheet(get_turf(src))
+
+/obj/machinery/bluespace_miner/process()
+	if(!check_factors())
+		return
+	spawn_mats()
+	playsound(src, 'sound/machines/ping.ogg', 50, FALSE)
+
+/obj/machinery/bluespace_miner/crowbar_act(mob/living/user, obj/item/tool)
+	if(default_deconstruction_crowbar(tool))
+		return TRUE
+
+/obj/machinery/bluespace_miner/screwdriver_act(mob/living/user, obj/item/tool)
+	. = TRUE
+	if(..())
+		return
+	if(default_deconstruction_screwdriver(user, icon_state, icon_state, tool))
+		update_appearance()
+		return
+	return FALSE
+
+/obj/item/circuitboard/machine/bluespace_miner
+	name = "Bluespace Miner (Machine Board)"
+	greyscale_colors = CIRCUIT_COLOR_GENERIC
+	build_path = /obj/machinery/bluespace_miner
+	req_components = list(
+		/obj/item/stack/sheet/glass = 1,
+		/obj/item/stack/ore/bluespace_crystal/refined = 1,
+		/obj/item/stock_parts/matter_bin = 2,
+		/obj/item/stock_parts/micro_laser = 2,
+		/obj/item/stock_parts/manipulator = 2)
+	needs_anchored = TRUE
+
+/datum/supply_pack/misc/bluespace_miner
+	name = "Bluespace Miner"
+	desc = "Nanotrasen has revolutionized the procuring of materials with bluespace-- featuring the Bluespace Miner!"
+	cost = CARGO_CRATE_VALUE * 2500 // 500,000, one of the most expensive buys (besides the bicycle, rip)
```

Comment:
I'm still not convinced that this is the right way to go about it, but we'll see.

I certainly don't want us to buff cargo so they make that much money every round, just so they can afford a bluespace miner, though.

Reply:
Yea, we can always change how to obtain it depending on how it performs.

Oh yea no, I do not want to buff cargo getting more money just to get these.